[INFRA] add steps for release protocol (PDF upload) (#455)
bids-standard/bids-specification: Release_Protocol.md

trigger [builds](https://readthedocs.org/projects/bids-specification/builds/)
for `stable` and the most recent tag.

### 9. Edit the mkdocs.yml file site_name to set a new development version

> Please submit a PR with the title `REL: <version>-dev`.
> This should be the first merged PR in the new version.

Reviewer: ```suggestion
Please commit to `master` with the title `REL: <version>-dev`.
This should be the first commit in the new version.
```

ref: https://github.com/bids-standard/bids-specification/pull/448#issuecomment-613681797

This is a small enough change we can trust this process
